test/e2e/storage/testsuites/testdriver.go

```diff
@@ -29,12 +29,32 @@ type TestDriver interface {
 	// GetDriverInfo returns DriverInfo for the TestDriver
 	GetDriverInfo() *DriverInfo
 	// CreateDriver creates all driver resources that is required for TestDriver method
-	// except CreateVolume
+	// except CreateVolume. May be called more than once and should only do something on
+	// the first call.
 	CreateDriver()
-	// CreateDriver cleanup all the resources that is created in CreateDriver
+	// CreateDriver cleanup all the resources that is created in CreateDriver. There is
+	// no guarantee that CreateDriver succeeded or even was called at all, so the test driver
+	// has to track resources.
 	CleanupDriver()
-	// SkipUnsupportedTest skips test in Testpattern is not suitable to test with the TestDriver
-	SkipUnsupportedTest(testpatterns.TestPattern)
+}
+
+// FilterTestDriver is an optional interface that drivers can
+// implement to filter out unsuitable tests while tests get defined.
+type FilterTestDriver interface {
+	// IsTestSupported returns false if the Testpattern is not
```

Comment:
Testpattern -> TestPattern

nit: "...returns *true* if the TestPattern is *suitable*..."

Reply:
Done.